app/services/idv/engine/events.rb

```diff
@@ -0,0 +1,277 @@
+    :result,
+    :billed,
+    :processed_alerts,
+    :alert_failure_count,
+    :address_line2_present,
+    keyword_init: true,
+  )
+
+  IdvFraudThreatmetrixCheckCompletedParams = Struct.new(
+    :request_success,
+    :request_timed_out,
+    :response_status,
+    :threatmetrix_session_id,
+    :threatmetrix_review_status,
+    keyword_init: true,
+  )
+
+  IdvFraudThreatmetrixCheckInitiatedParams = Struct.new(
+    :threatmetrix_session_id,
+    keyword_init: true,
+  )
+
+  IdvUserEnteredPasswordParams = Struct.new(
+    :password,
+    keyword_init: true,
+  )
+
+  IdvUserEnteredSsnParams = Struct.new(
+    :ssn,
+    keyword_init: true,
+  )
+
+  IdvUserUpdatedMailingAddressParams = Struct.new(
+    :address,
+    keyword_init: true,
+  )
+
+  # The user has changed their password.
+  # @param [Object] params
+  # @return [AuthUserChangedPasswordParams]
+  def auth_user_changed_password(params)
+    raise 'params cannot be nil' if params.nil?
+
+    if params.is_a?(Hash)
+      params = AuthUserChangedPasswordParams.new(params)
+    end
+
+    handle_event :auth_user_changed_password, params
+    params
+  end
+
+  # The user has reset their password and will not be able to access their IDV
+  # PII until providing their personal key.
+  # @return [nil]
+  def auth_user_reset_password
+    handle_event :auth_user_reset_password
+    nil
+  end
+
+  # The user's address was submitted to LexisNexis InstantVerify.
+  # @return [nil]
+  def idv_address_submitted_to_instantverify
+    handle_event :idv_address_submitted_to_instantverify
+    nil
+  end
+
+  # The user has uploaded identity documents, and Login.gov has submitted them
+  # to Acuant for  processing.
+  # @param [Object] params
+  # @return [IdvDocumentsSubmittedToAcuantParams]
+  def idv_documents_submitted_to_acuant(params)
+    raise 'params cannot be nil' if params.nil?
+
+    if params.is_a?(Hash)
+      params = IdvDocumentsSubmittedToAcuantParams.new(params)
+    end
+
+    handle_event :idv_documents_submitted_to_acuant, params
+    params
+  end
+
+  # The user has uploaded identity documents, and Login.gov has submitted them
+  # to LexisNexis  TrueID for processing.
+  # @param [Object] params
+  # @return [IdvDocumentsSubmittedToTrueidParams]
+  def idv_documents_submitted_to_trueid(params)
+    raise 'params cannot be nil' if params.nil?
+
+    if params.is_a?(Hash)
+      params = IdvDocumentsSubmittedToTrueidParams.new(params)
+    end
+
+    handle_event :idv_documents_submitted_to_trueid, params
+    params
+  end
+
+  # Login.gov requested the result of an automated fraud check.
+  # @param [Object] params
+  # @return [IdvFraudThreatmetrixCheckCompletedParams]
+  def idv_fraud_threatmetrix_check_completed(params)
+    raise 'params cannot be nil' if params.nil?
+
+    if params.is_a?(Hash)
+      params = IdvFraudThreatmetrixCheckCompletedParams.new(params)
+    end
+
+    handle_event :idv_fraud_threatmetrix_check_completed, params
+    params
+  end
```

Comment:
looking at this file, I'm not sure the auto codegen on it is worth the effort to maintain? It seems like that if we needed custom code, we'd have to add things like the `Verification` class anyways?

Reply:
Yeah, I'm not 100% sold on it, this was just to get some stuff bootstrapped. My favorite thing about codegen as a strategy is that you can always just delete the generator and maintain what it generated